comit/src/expiries.rs

```diff
@@ -0,0 +1,591 @@
+            return acc;
+        }
+
+        let (_action, next_state) = state.next();
+        let transition_period = state.transition_period(config);
+
+        period_to_complete(config, next_state, acc + transition_period)
+    }
+
+    period_to_complete(config, current_state, Duration::zero())
+}
+
+/// Happy path states for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceState {
+    /// Initial state, swap not yet created.
+    None,
+    /// Swap has been created.
+    Created,
+    /// The fund alpha transaction has been sent to the network.
+    FundAlphaTransactionSent,
+    /// Implies fund alpha transaction has reached finality.
+    AlphaFunded,
+    /// Implies fund beta transaction has reached finality.
+    BetaFunded,
+    /// The redeem beta transaction has been sent to the network.
+    RedeemBetaTransactionSent,
+    /// Implies beta redeem transaction has reached finality.
+    Done,
+}
+
+/// Possible next action for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceAction {
+    // Happy path actions for Alice.
+    Create,
+    FundAlpha,
+    WaitForFundTransactionFinality,
+    WaitForBetaFunded,
+    RedeemBeta,
+    WaitForRedeemBetaTransactionFinality,
+    NoFurtherAction, // Implies swap is done from Alice's perspective.
+    // Cancel path.
+    Abort,         // Implies HTLC not funded, abort but take no further action.
+    WaitForRefund, // Implies alpha ledger HTLC funded but expiry not yet elapsed.
+    Refund,        // Implies HTLC funded and expiry time elapsed.
+}
+
+impl AliceState {
+    /// Gets the next action required to transition to the next state.
+    fn next(&self) -> (AliceAction, AliceState) {
+        use self::{AliceAction::*, AliceState::*};
+
+        match self {
+            None => (Create, Created),
+            Created => (FundAlpha, FundAlphaTransactionSent),
+            FundAlphaTransactionSent => (WaitForFundTransactionFinality, AlphaFunded),
+            AlphaFunded => (WaitForBetaFunded, BetaFunded),
+            BetaFunded => (RedeemBeta, RedeemBetaTransactionSent),
+            RedeemBetaTransactionSent => (WaitForRedeemBetaTransactionFinality, Done),
+            Done => (NoFurtherAction, Done),
+        }
+    }
+
+    /// The minimum time we need to allow to transition to the next state.
+    fn transition_period(&self, c: &Config) -> Duration {
+        use self::AliceAction::*;
+
+        let (next_action, _next_state) = self.next();
+
+        match next_action {
+            Create => {
+                // Transition from None to Created
+                c.alice_to_act()
+            }
+            FundAlpha => {
+                // Transition from Created to FundAlphaTransactionSent
+                c.alice_to_act()
+            }
+            WaitForFundTransactionFinality => {
+                // Transition from FundAlphaTransactionSent to AlphaFunded
+                c.include_alpha_fund_transaction() + c.finality_alpha()
```

Comment:
```
c.include_alpha_fund_transaction()
```
I am guessing this is the time we assume it would take to mine the transaction? `mine` terminology would be better. Also, this is not specific to a `fund` transaction, or is it?

Reply:
Cool, you and @thomaseizinger now both prefer 'mine' over 'include' - will use. Yes this is specific to the fund, I see it as possible, in the future, that we may want to influence the mine latency of the fund transaction differently to the redeem transaction (lower fees for fund for example).